### ISSUE TITLE
Wrapping unicode characters causes Timewarrior to hang indefinitely

### ISSUE THREAD

**Issue by lauft** [GothenburgBitFactory/libshared#23]
The following lines will cause Timewarrior 1.3.0 to hang indefinetely (https://github.com/GothenburgBitFactory/timewarrior/issues/309):
```
$ src/timew track 2016-01-16T08:55:00 - 2016-01-16T09:00:00 "안녕하세요 월드"
Note: '"안녕하세요 월드"' is a new tag.
Recorded "안녕하세요 월드"
  Started 2016-01-16T08:55:00
  Ended              09:00:00
  Total               0:05:00
$ src/timew day 2016-01-16 - 2016-01-17
```
I can reproduce this by adding the following test to `shared.t.cpp:65`:
```
  text = "안녕하세요 월드";
  lines.clear ();
  wrapText (lines, text, 1, false);
``` 
This test passes if the `width` parameter in `wrapText(...)` gets set to `2`:
```
  text = "안녕하세요 월드";
  lines.clear ();
  wrapText (lines, text, 2, false);
  t.is (lines.size (), (size_t) 7, "wrapText '안녕하세요 월드' -> 8 lines");
  t.is (lines[0], "안", "wrapText line 0 -> '안'");
  t.is (lines[1], "녕", "wrapText line 1 -> '녕'");
  t.is (lines[2], "하", "wrapText line 2 -> '하'");
  t.is (lines[3], "세", "wrapText line 3 -> '세'");
  t.is (lines[4], "요", "wrapText line 4 -> '요'");
  //t.is (lines[5], " ", "wrapText line 5 -> ' '");
  t.is (lines[5], "월", "wrapText line 6 -> '월'");
  t.is (lines[6], "드", "wrapText line 7 -> '드'");
```
However, the whitespace in `안녕하세요 월드` gets omitted somehow, so there is not empty `line 5`

**Comment by wodny** [comment]
Hi, please check if the patch from the PR is good enough.

**Comment by lauft** [comment]
Fixed with #28
